daemon/config/config.go

```diff
@@ -398,7 +398,7 @@ func getConflictFreeConfiguration(configFile string, flags *pflag.FlagSet) (*Con
 	if flags != nil {
 		var jsonConfig map[string]interface{}
 		reader = bytes.NewReader(b)
-		if err := json.NewDecoder(reader).Decode(&jsonConfig); err != nil {
+		if err := json.NewDecoder(reader).Decode(&jsonConfig); err != nil && err != io.EOF {
```

Comment:
what change caused this? Also, this ignores the errors after first json block. As the data is already in byte slice then `Unmarshal` is cleaner.

Reply:
I think this was added to allow for an empty `daemon.json` file (empty file, not empty `{}`)

Reply:
Not sure if this is a good idea. But if added should be an explicit empty check then. Tbh I'm surprised this returns `io.EOF` and not `ErrUnexpectedEOF`.

Reply:
I think it would return `ErrUnexpectedEOF` if the JSON is incomplete (`{`), and `EOF` if the file is empty (or only whitespace);

https://play.golang.org/p/1_N1QidcM7T

```go
package main

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io"
)

func main() {
	var reader io.Reader
	var jsonConfig map[string]interface{}

	for _, config := range []string{"{", "[", `""`, "", "     ", `[]`, `{}`} {
		reader = bytes.NewReader([]byte(config))

		if err := json.NewDecoder(reader).Decode(&jsonConfig); err != nil {
			fmt.Println(err)
			continue
		}
		fmt.Println(config, "is ok")

	}
}
```

Prints:

```
unexpected EOF
unexpected EOF
json: cannot unmarshal string into Go value of type map[string]interface {}
EOF
EOF
json: cannot unmarshal array into Go value of type map[string]interface {}
{} is ok
```

> Not sure if this is a good idea

I thought slightly relaxing would be ok; we already did allow an empty `{}`; considering an empty file to be the equivalent (config with nothing set).

Do you have specific concerns?

Reply:
Yeah, idk why empty and spaces are considered as valid json for `Decode`. `JSON.parse` in js in unexpectedeof for both cases.

I'm not strictly against allowing empty file, if detection for it is clear in code. It might lead to error going unnoticed though. If Desktop fills this file from a field I do think correct behavior would be to not create a file if the field is empty, not to create an empty file.